### PR TITLE
Hide EditText's cursor when keyboard is hidden and vice versa

### PR DESCRIPTION
**Pull request by tuancoltech**
- Hide EditText's cursor when keyboard is hidden and vice versa: https://app.asana.com/0/1207819682524134/1208211226661165
- [enhance] Pasting a text should append it to the current cursor position in Text note: https://app.asana.com/0/1207819682524134/1208091975005939

### CODE REVIEW

**Review by hieuwu — APPROVED** [review]

**Comment by tuancoltech** [comment]
Hey, @hieuwu @shubertm 
I pushed one more commit which is closely related to the original target of this PR.
Now, both commits are enhancement of the Text note edit page.
Could you help review again?

Thanks.

File: app/src/main/java/dev/arkbuilders/arkmemo/ui/fragments/EditTextNotesFragment.kt [hunk 1]
```diff
@@ -28,9 +32,23 @@ class EditTextNotesFragment: BaseEditNoteFragment() {
...
+                val newCursorPos = if (cursorPos < 0) {
+                    (clipBoardText.length + noteContent.length - 1).coerceAtLeast(0)
+                } else {
+                    clipBoardText.length + cursorPos
+                }
+                try {
+                    newTextBuilder.append(noteContent.insertStringAtPosition(clipBoardText, cursorPos))
+                } catch (e: IndexOutOfBoundsException) {
+                    Log.e(TAG, "pasteNoteClickListener exception: ${e.message}")
+                    newTextBuilder.append(noteContent).append(clipBoardText)
```
Comment (shubertm) [inline app/src/main/java/dev/arkbuilders/arkmemo/ui/fragments/EditTextNotesFragment.kt:47]: Does this mean, pasting is not always dependent on cursor position?

**Review by shubertm — COMMENTED** [review]
Overall, it is working. Good job @tuancoltech

File: app/src/main/java/dev/arkbuilders/arkmemo/ui/fragments/EditTextNotesFragment.kt (hunk 1 above)
Comment (tuancoltech) [reply app/src/main/java/dev/arkbuilders/arkmemo/ui/fragments/EditTextNotesFragment.kt:47]: @shubertm In theory pasting should be dependent on cursor position, but if there's a case where the cursor position is invalid (e.g, cursor is not found anywhere within the text), then the pasted content should be appended to the end of the note's text.

**Review by hieuwu — APPROVED** [review]

**Pull request merged**